File: api/internal/domain/models/task.go
package models

import (
	"HITS_ToDoList_Tests/internal/domain/enums"
	"github.com/google/uuid"
	"time"
)

type Task struct {
	ID          uuid.UUID
	CreatedAt   time.Time
	ChangedAt   *time.Time
	Name        string
	Description *string
	Deadline    *time.Time
	Status      enums.Status
	Priority    enums.Priority
}

func NewTask(name string, description *string, deadline *time.Time, status *enums.Status,
	priority *enums.Priority) *Task {
	task := &Task{
		ID:          uuid.New(),
		CreatedAt:   time.Now(),
		Name:        name,
		Description: description,
		Deadline:    deadline,
	}

	if status == nil {
		task.Status = enums.Active
	} else {
		task.Status = *status
	}

	if priority == nil {
		task.Priority = enums.Medium
	} else {
		task.Priority = *priority
	}

	return task
}

File: frontend/src/entities/task.ts
import type { priority } from "../enums/priority.ts";
import type { status } from "../enums/status.ts";

export class task {
    public id: string;
    public createdAt: Date;
    public changedAt?: Date;
    public name: string;
    public description?: string;
    public deadline?: Date;
    public priority: priority;
    public status: status;
    public isDone: boolean;

    constructor(id: string, name: string, priority: priority, status: status, createdAt: Date, changedAt?: Date, description?: string, deadline?: Date) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.deadline = deadline;
        this.priority = priority;
        this.status = status;
        this.createdAt = createdAt;
        this.changedAt = changedAt;

        this.isDone = status === "Completed" || status === "Late";
    }
}